### PR TITLE
Open issue: numeric pseudo-inverse algorithm

### PR DESCRIPTION
**Pull request by 1fisedi**
Added new algorithm for pseudo-inverse using moore-penrose algorithms and SVD.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Numeric/Pseudoinverse/PseudoInverse.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /// </summary>
+    public static class PseudoInverse
+    {
+        /// <summary>
+        /// Return the pseudoinverse of a matrix based on the Moore-Penrose Algorithm.
+        /// using Singular Value Decomposition (SVD).
+        /// </summary>
+        /// <param name="inMat">Input matrix to find its inverse to.</param>
+        /// <returns>The inverse matrix approximation of the input matrix.</returns>
+        public static double[,] PInv(double[,] inMat)
```
Comment (siriak) [inline Algorithms/Numeric/Pseudoinverse/PseudoInverse.cs:21]: Please make variable names more readable

**Review by siriak — COMMENTED** [review]
Thanks for the algorithm, see my comments below

File: Algorithms.Tests/Numeric/PseudoInverse/PseudoInverseTests.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            var inMatCopy = new double[,] { { 2, 4, 6 }, { 2, 0, 2 }, { 6, 8, 14 } };
+
+            // Act
+            var result = PseudoInverse.PInv(inMat);
+            var aainva = inMatCopy.Multiply(result).Multiply(inMatCopy);
+
+            var rounded = aainva.RoundToNextInt();
+            var isequal = rounded.IsEqual(inMatCopy);
+            // Assert
+            Assert.IsTrue(isequal);
```
[the remark this answers is not in the document]
Comment (1fisedi) [reply Algorithms.Tests/Numeric/PseudoInverse/PseudoInverseTests.cs:24]: I added two ways to check, one based on AA+A = A; and another using (A+)+ = A. That applies to all matrices, independent of their rank.

File: Utilities/Extensions/VectorExtensions.cs [hunk 3]
```diff
@@ -131,5 +131,44 @@ public static double[] ToRowVector(this double[,] source)
 
             return rowVector;
         }
+
+        /// <summary>
+        /// Calculates the reciprocal of every element in the vector.
+        /// </summary>
+        /// <param name="vector">The vector.</param>
+        /// <returns>The vector with calculated reciprocals of elements.</returns>
+        public static double[] Reciprocal(this double[] vector)
```
[the remark this answers is not in the document]
Comment (1fisedi) [reply Utilities/Extensions/VectorExtensions.cs]: removed.

**Pull request merged**